train_segmentation.py

```diff
@@ -438,7 +438,7 @@ def evaluation(eval_loader,
                                    ep_num=ep_idx + 1,
                                    scale=scale)
 
-            loss = criterion(outputs, labels.unsqueeze(1).float())
+            loss = criterion(outputs, labels) if num_classes > 1 else criterion(outputs, labels.unsqueeze(1).float())
```

Comment:
have you tested that this line fails for the multiclass evaluation? 
In my tests loss = criterion(outputs, labels.unsqueeze(1).float()) works for both cases

Reply:
yeah I created a test for it, it dont crash for all binary but some yeah and it pass for the news tests for binary and multiclass loss function that have a yaml in config